pkg/kubelet/kubelet.go

```diff
@@ -285,24 +285,21 @@ func makePodSourceConfig(kubeCfg *kubeletconfiginternal.KubeletConfiguration, ku
 }
 
 // PreInitRuntimeService will init runtime service before RunKubelet.
-func PreInitRuntimeService(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
-	kubeDeps *Dependencies,
-	remoteRuntimeEndpoint string,
-	remoteImageEndpoint string) error {
-	// remoteImageEndpoint is same as remoteRuntimeEndpoint if not explicitly specified
-	if remoteRuntimeEndpoint != "" && remoteImageEndpoint == "" {
-		remoteImageEndpoint = remoteRuntimeEndpoint
+func PreInitRuntimeService(kubeCfg *kubeletconfiginternal.KubeletConfiguration, kubeDeps *Dependencies) error {
+	// ImageServiceEndpoint is same as ContainerRuntimeEndpoint if not explicitly specified
+	if kubeCfg.ContainerRuntimeEndpoint != "" && kubeCfg.ImageServiceEndpoint == "" {
+		kubeCfg.ImageServiceEndpoint = kubeCfg.ContainerRuntimeEndpoint
```

Comment:
Is this necessary, given your addition in [pkg/kubelet/apis/config/v1beta1/defaults.go](https://github.com/kubernetes/kubernetes/pull/112136/files#diff-64312739de7a7da86a275e2ada9e0caf61e826c20b90ffbb6c1bafa464d15fcf)

Reply:
not that necessary， but it can be like this（IMO）. this will make reader clear to know both are set.

I will update here tomorrow if it has to（from my phone）

Reply:
Sorry, I looked at this comment on my phone and understand it wrongly. This may be removed as we did it in `defaults.go`. Let me clean-up this.